src/renderer/features/wallet-details/ui/wallets/VaultWalletDetails.tsx

```diff
@@ -79,21 +85,23 @@ export const VaultWalletDetails = ({ wallet, onClose }: Props) => {
     setChains(filteredChains);
   }, []);
 
-  const handleConstructorKeys = (
-    keysToAdd: (VaultChainAccount | VaultShardAccount[])[],
-    keysToRemove: (VaultChainAccount | VaultShardAccount[])[],
-  ) => {
+  const handleConstructorKeys = (keys: DerivationKeyDraft[]) => {
     toggleConstructorModal();
 
+    const draftKeySet = new Set(keys.map(k => k.chainId + k.derivationPath));
+    const existingKeySet = new Set(wallet.accounts.map(a => a.chainId + a.derivationPath));
```

Comment:
Better to create a function and avoid duplicating code
```suggestion
const createKeySet = (items: Array<{ chainId: string; derivationPath: string }>) => 
  new Set(items.map(item => item.chainId + item.derivationPath));

const draftKeySet = createKeySet(keys);
const existingKeySet = createKeySet(wallet.accounts);
```